test/e2e/atlas/cluster_delete_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//go:build e2e || (atlas && clusters && terminationProtection)
+
+package atlas_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/mongodb/mongodb-atlas-cli/test/e2e"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+func TestClusterDeletion(t *testing.T) {
```

Comment:
[nit] since creating a cluster takes a lot of time, instead of adding a new tests to create a cluster, let's reuse one of the test we have, for example 
https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/e2e/atlas/clusters_flags_test.go, either update the create of that test to enable the protection or add an update step to enable it after creation

Reply:
Sounds good to me :3